### PR TITLE
Azure security rules plugin tests

### PR DESCRIPTION
**Pull request by jadsonluan**

### CODE REVIEW

**Comment by fgan1** [comment]
It seems that the common(develop branch) dependency is conflicting. Which branch or commit is compatible?

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/securityrule/AzureSecurityRulePlugin.java [hunk 1]
```diff
@@ -82,7 +82,7 @@ public String requestSecurityRule(SecurityRule securityRule, Order majorOrder, A
     public void deleteSecurityRule(String securityRuleId, AzureUser azureUser) throws FogbowException {
         LOGGER.info(String.format(Messages.Info.DELETING_INSTANCE_S, securityRuleId));
 
-        SecurityRuleIdContext securityRuleIdContext = new SecurityRuleIdContext(securityRuleId);
+        SecurityRuleIdContext securityRuleIdContext = this.getSecurityRuleIdContext(securityRuleId);
```
Comment (wassoares) [inline src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/securityrule/AzureSecurityRulePlugin.java:85]: The changes made to this class should be in the azure-security-rules-plugin branch and not in this testing branch.

**Comment by jadsonluan** [comment]
> It seems that the common(develop branch) dependency is conflicting. Which branch or commit is compatible?

The problem is solved. The branch was outdated in relation to the develop.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/securityrule/sdk/AzureNetworkSecurityGroupOperationSDKTest.java [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+//        Mockito.when(networkSecurityRule.direction()).thenReturn(securityRuleDirection);
+//        Mockito.when(networkSecurityRule.destinationPortRange()).thenReturn(portRange);
+//        Mockito.when(networkSecurityRule.protocol()).thenReturn(securityRuleProtocol);
+//        Mockito.when(networkSecurityRule.inner()).thenReturn(securityRuleInner);
+//
+//        Mockito.when(AzureSecurityRuleUtil.getFogbowDirection(Mockito.eq(securityRuleDirection))).thenReturn(direction);
+//        Mockito.when(AzureSecurityRuleUtil.getPorts(Mockito.eq(portRange))).thenReturn(ports);
+//        Mockito.when(AzureSecurityRuleUtil.getIpAddress(Mockito.eq(cidr))).thenReturn(ipAddress);
+//        Mockito.when(AzureSecurityRuleUtil.inferEtherType(Mockito.eq(ipAddress))).thenReturn(etherType);
+//        Mockito.when(AzureSecurityRuleUtil.getFogbowProtocol(Mockito.eq(securityRuleProtocol.toString()))).thenReturn(protocol);
```
[the remark this answers is not in the document]
Comment (jadsonluan) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/securityrule/sdk/AzureNetworkSecurityGroupOperationSDKTest.java]: Done.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/securityrule/sdk/AzureNetworkSecurityGroupOperationSDKTest.java [hunk 3]
```diff
@@ -0,0 +1,446 @@
...
+
+        String ruleName = SystemConstants.FOGBOW_INSTANCE_NAME_PREFIX + "test-security-rule";
+        String networkSecurityGroupName = "network-security-group-name";
+        String networkSecurityGroupId = "network-security-group-id";
+        String cidr = TestUtils.DEFAULT_CIDR;
+        int portFrom = DEFAULT_PORT_FROM;
+        int portTo = DEFAULT_PORT_TO;
+
+        SecurityRule.Direction direction = SecurityRule.Direction.IN;
+        SecurityRuleDirection securityRuleDirection = SecurityRuleDirection.INBOUND;
```
[the remark this answers is not in the document]
Comment (jadsonluan) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/securityrule/sdk/AzureNetworkSecurityGroupOperationSDKTest.java]: Done.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/securityrule/AzureSecurityRulePluginTest.java [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({
+        AzureGeneralUtil.class,
+        SecurityRuleIdContext.class
+})
+public class AzureSecurityRulePluginTest extends TestUtils {
+    private static final int DEFAULT_PORT_FROM = 22;
+    private static final int DEFAULT_PORT_TO = 22;
+    private AzureSecurityRulePlugin plugin;
+    private Azure azure;
```
[the remark this answers is not in the document]
Comment (jadsonluan) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/securityrule/AzureSecurityRulePluginTest.java]: Done. I removed only the Azure attribute which is the only unused

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/securityrule/AzureSecurityRulePluginTest.java [hunk 5]
```diff
@@ -0,0 +1,213 @@
+package cloud.fogbow.ras.core.plugins.interoperability.azure.securityrule;
+
+import cloud.fogbow.common.exceptions.FogbowException;
+import cloud.fogbow.common.exceptions.InvalidParameterException;
```
[the remark this answers is not in the document]
Comment (jadsonluan) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/securityrule/AzureSecurityRulePluginTest.java]: Removed.

**Pull request merged**